From ba7e35f6237f814f47bcca911deada2062aaa2e0 Mon Sep 17 00:00:00 2001
From: Jason Rogena <j.rogena@cgiar.org>
Date: Wed, 2 Apr 2014 09:08:27 +0300
Subject: [PATCH 140/213] Version 0.8.15: Feature #146 implemented. Keyboard no
 longer shows when date EditText comes into focus. Keyboard also stays hidden
 by default when activity resumes

---
 MistroFarmer/src/main/AndroidManifest.xml                     |  5 +++--
 .../cgiar/ilri/mistro/farmer/AddMilkProductionActivity.java   |  5 +++++
 .../java/org/cgiar/ilri/mistro/farmer/LandingActivity.java    | 11 ++++++++++-
 3 files changed, 18 insertions(+), 3 deletions(-)

diff --git a/MistroFarmer/src/main/AndroidManifest.xml b/MistroFarmer/src/main/AndroidManifest.xml
index 6c99b82..8ab3278 100644
--- a/MistroFarmer/src/main/AndroidManifest.xml
+++ b/MistroFarmer/src/main/AndroidManifest.xml
@@ -2,7 +2,7 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="org.cgiar.ilri.mistro.farmer"
     android:versionCode="2"
-    android:versionName="0.8.13" >
+    android:versionName="0.8.15" >
 
     <uses-sdk
         android:minSdkVersion="7"
@@ -78,7 +78,8 @@
         </activity>
         <activity
             android:name="org.cgiar.ilri.mistro.farmer.AddMilkProductionActivity"
-            android:label="@string/title_activity_add_milk_production" >
+            android:label="@string/title_activity_add_milk_production"
+            android:windowSoftInputMode="stateHidden" >
         </activity>
     </application>
 
diff --git a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddMilkProductionActivity.java b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddMilkProductionActivity.java
index b4b64bb..69a915d 100644
--- a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddMilkProductionActivity.java
+++ b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/AddMilkProductionActivity.java
@@ -14,6 +14,7 @@ import android.telephony.TelephonyManager;
 import android.util.Log;
 import com.actionbarsherlock.view.Menu;
 import android.view.View;
+import android.view.inputmethod.InputMethodManager;
 import android.widget.ArrayAdapter;
 import android.widget.Button;
 import android.widget.DatePicker;
@@ -78,6 +79,10 @@ public class AddMilkProductionActivity extends SherlockActivity implements View.
         cowS=(Spinner)this.findViewById(R.id.cow_s);
         dateTV=(TextView)this.findViewById(R.id.date_tv);
         dateET=(EditText)this.findViewById(R.id.date_et);
+
+        InputMethodManager inputMethodManager = (InputMethodManager) this.getSystemService(Context.INPUT_METHOD_SERVICE);
+        inputMethodManager.hideSoftInputFromWindow(dateET.getWindowToken(), 0);
+
         dateET.setOnClickListener(this);
         timeTV=(TextView)this.findViewById(R.id.time_tv);
         timeS=(Spinner)this.findViewById(R.id.time_s);
diff --git a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/LandingActivity.java b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/LandingActivity.java
index 0950ca6..f38ab36 100644
--- a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/LandingActivity.java
+++ b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/LandingActivity.java
@@ -6,6 +6,8 @@ import android.app.ProgressDialog;
 import android.content.Context;
 import android.content.DialogInterface;
 import android.content.Intent;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
 import android.os.AsyncTask;
 import android.os.Bundle;
 import android.telephony.TelephonyManager;
@@ -77,7 +79,14 @@ public class LandingActivity extends SherlockActivity implements View.OnClickLis
 
         //init text according to locale
         initTextInViews();
-        Toast.makeText(this, "Version 0.8 Revision 13", Toast.LENGTH_LONG).show();
+
+        //get version name for the application and toast it
+        try {
+            PackageInfo pInfo = getPackageManager().getPackageInfo(getPackageName(), 0);
+            Toast.makeText(this, "Version "+pInfo.versionName, Toast.LENGTH_LONG).show();
+        } catch (PackageManager.NameNotFoundException e) {
+            e.printStackTrace();
+        }
     }
 
     public boolean onCreateOptionsMenu(Menu menu) {
-- 
2.7.1

